[kineto] populate src/dst rank for p2p

shengbao-zheng opened the pull request:
Summary:
as title
populate src/dst rank (global rank) for p2p kernel

Differential Revision: D59794535

aaronenyeshi reviewed (approved):
LGTM, please fix lintrunner and unit tests

shengbao-zheng committed:
[profiler]populate src/dst rank for p2p (#130812)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/130812

as title
populate src/dst rank (global rank) for p2p kernel

Reviewed By: aaronenyeshi, minsii

Differential Revision: D59794535
